### ISSUE TITLE
Allow using attribute replacement in theme keys that allow arbitrary text

### ISSUE THREAD

**Issue by DavidGamba** [asciidoctor/asciidoctor-pdf#588]
I have the following `footer` code, that uses attributes and is working properly:

```
footer:
   recto: # odd pages
      content:
         left: '{page-number}'
         center: '©2016 {company} v{docversion}, {docdate}'
         right: 'image:{base_dir}/images/common/logo.png[width=150]'
```

However, when I try the same in the `title_page`, the attributes are not replaced:

```
title_page:
   logo_image:  '{base_dir}/images/{product-short}/{product-short}_logo.png'
```

In the `title_page_logo_image`'s case I am forced to use the full path of the image because attribute replacements do not work. Attribute replacements should be enabled in all keys that allow text or file inputs.

**Comment by mojavelinux** [comment]
You are correct that attributes are only replaced for running content atm. I think it's fair to assume that we can add attribute substitutions for images (which are the only file inputs) and content keys.

--

Btw, I recommend switching to the new running content structure:

```yaml
footer:
  recto:
    left:
      content: '{page-number}'
```

Among other reasons, one reason we did this was so that we could more easily find the content key (as it should be the last segment, i.e., the property name).

**Comment by mojavelinux** [comment]
I looked into how feasible it would be to give the theme access to the document attributes. What I determined is that it would couple the theme to strongly with the document and would add a lot of complexity to the theme loader. So I don't think that's the right approach.

A better solution is to apply attribute substitutions at the point the theme key is read. In fact, that's how the content key of the running content works already (so it builds on precedent and existing code). We can start by replacing attributes in the value of image keys. We can then address additional use cases in separate issues.
